### PR TITLE
#2 Metrics and Statistics

### LINKED ISSUE #2: Metrics and Statisitcs
Provide a API to access Æternitas' metrics

## Metrics stored in redis
Grouped by pollable class
 - [x]  job execution count 
 - [x]  failed job count, failure_ratio
 - [x]  ignored errors
 - [x]  deactivations
 - [x]  execution time
 - [x] guard
 - [x] waited
 - [x]  guard wait sleep
 - [x]  guard timeout exceeded

## Data statistics
 - [ ] number of sources (per pollable class)
 - [ ] pollables created (per pollable class)
 - [ ] future pollings (per pollable class)

### PR DESCRIPTION
**Pull request by DarthMax**
Fixes #2 
Implements polling statistic API

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,3 +2,5 @@ source 'https://rubygems.org'
 
 # Specify your gem's dependencies in aeternitas.gemspec
 gemspec
+
+gem 'tabs', git: 'https://github.com/DarthMax/tabs', branch: 'relax-redis-requirement'
```
Comment (tamaloa) [inline Gemfile]: We should use our tabstabs 2.0 gem for this PR

File: lib/aeternitas/metrics.rb [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  # Provides extensive metrics for Aeternitas.
+  # Every metric is scoped by pollable class.
+  # Available metrics are:
+  #   - polls => Number of polling runs
+  #   - successful_polls => Number of successful polling runs
+  #   - failed_polls => Number of failed polling runs (includes IgnoredErrors,
+  #     excludes deactivation errors and Lock errors)
+  #   - ignored_errors => Number of raised {Aeternitas::Errors::Ignored}
+  #   - deactivation_errors => Number of errors raised which are declared as deactivation_errors
+  #   - execution_time => Job execution time in seconds
```
Comment (tamaloa) [inline lib/aeternitas/metrics.rb]: average or total ?

File: lib/aeternitas/pollable.rb [hunk 3]
```diff
@@ -39,6 +39,8 @@ module Pollable
 
       before_validation ->(pollable) { pollable.pollable_meta_data ||= pollable.build_pollable_meta_data(state: 'waiting' ); true }
 
+      after_create ->(pollable) { Aeternitas::Metrics.log(:pollables_created, pollable) }
```
Comment (tamaloa) [inline lib/aeternitas/pollable.rb]: should we do this on_commit?

File: lib/aeternitas/metrics.rb (hunk 2 above)
Comment (DarthMax) [inline lib/aeternitas/metrics.rb]: both ;)

**Pull request merged**